src/helpers/environment-helper.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Yoast\WP\SEO\Helpers;
+
+/**
+ * A helper object for site environment.
+ */
+class Environment_Helper {
+
+	/**
+	 * Determines if the site is running on production.
+	 *
+	 * @return bool True if WordPress is currently running on production, false for all other environments.
+	 */
+	public function is_production_mode() {
+		return $this->get_wp_environment() === 'production';
```

Comment:
I would call `\wp_get_environment_type()` instead of wrapping the method that gets mocked. Just mock `wp_get_environment_type`.